File: wrappers/dotnet/src/BbsSignatures/Models/CreateBlindedCommitmentRequest.cs
using System;

namespace BbsSignatures
{
    /// <summary>
    /// A request to create a BBS signature that features blinded/committed messages
    /// </summary>
    public class CreateBlindedCommitmentRequest
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="CreateBlindedCommitmentRequest"/> class.
        /// </summary>
        /// <param name="publicKey">The public key.</param>
        /// <param name="messages">The messages.</param>
        /// <param name="nonce">The nonce.</param>
        /// <exception cref="ArgumentNullException">
        /// publicKey
        /// or
        /// messages
        /// or
        /// nonce
        /// </exception>
        public CreateBlindedCommitmentRequest(BbsKey publicKey, IndexedMessage[] messages, byte[] nonce)
        {
            Key = publicKey ?? throw new ArgumentNullException(nameof(publicKey));
            Messages = messages ?? throw new ArgumentNullException(nameof(messages));
            Nonce = nonce ?? throw new ArgumentNullException(nameof(nonce));
        }

        /// <summary>
        /// The public key
        /// </summary>
        public BbsKey Key { get; set; }

        /// <summary>
        /// The known messages to sign
        /// </summary>
        public IndexedMessage[] Messages { get; set; }

        /// <summary>
        /// A nonce for the resulting proof
        /// </summary>
        public byte[] Nonce { get; set; }
    }
}

File: wrappers/python/ursa_bbs_signatures/models/CreateBlindedCommitmentRequest.py
from typing import List
from .keys import BbsKey
from .IndexedMessage import IndexedMessage


class CreateBlindedCommitmentRequest:

    def __init__(self, public_key: BbsKey, messages: List[IndexedMessage], nonce: bytes) -> None:
        """A request to create a BBS signature that features blinded/committed messages

        Args:
            public_key: The public key
            messages: The messages
            nonce: The nonce
        """
        self.public_key = public_key
        self.messages = messages
        self.nonce = nonce
